ReactAndroid/src/main/java/com/facebook/react/bridge/ReadableNativeMap.java

```diff
@@ -48,6 +53,35 @@ public ReadableMapKeySetIterator keySetIterator() {
     return new ReadableNativeMapKeySetIterator(this);
   }
 
+  public HashMap<String, Object>toHashMap() throws ReadableTypeException {
+    ReadableMapKeySetIterator iterator = keySetIterator();
+    HashMap<String, Object> hashMap = new HashMap<>();
+
+    while (iterator.hasNextKey()) {
+      String key = iterator.nextKey();
+      switch (getType(iterator.nextKey())) {
```

Comment:
shouldn't `iterator.nextKey()` be here `key` ? Wouldn't this consume another key ?